### DISCUSSION TITLE
Does vcpkg need to install Visual Studio on a Windows system?

### DISCUSSION THREAD

**Discussion by maxpeng** [microsoft/vcpkg#35379] [Q&A]
I tried this tutorial - [Tutorial: Install and use packages with vcpkg](https://learn.microsoft.com/en-us/vcpkg/get_started/get-started?pivots=shell-cmd) on a Windows 10 system with msys2 installed but without Visual Studio, I got following error after these two commands ```export VCPKG_DEFAULT_HOST_TRIPLET=x64-mingw-static``` and ```cmake --preset=default```:

```
$ cmake --preset=default
Preset CMake variables:

  CMAKE_TOOLCHAIN_FILE="C:\...\vcpkg\scripts\buildsystems\vcpkg.cmake"

-- Building for: NMake Makefiles
-- Running vcpkg install
Fetching registry information from https://github.com/microsoft/vcpkg (HEAD)...
Detecting compiler hash for triplet x64-mingw-static...
error: in triplet x64-windows: Unable to find a valid Visual Studio instance
Could not locate a complete Visual Studio instance

-- Running vcpkg install - failed
CMake Error at C:/.../vcpkg/scripts/buildsystems/vcpkg.cmake:899 (message):
  vcpkg install failed.  See logs for more information:
  C:\...\vcpkg\helloworld\build\vcpkg-manifest-install.log
Call Stack (most recent call first):
  C:/.../cmake/share/cmake-3.24/Modules/CMakeDetermineSystem.cmake:124 (include)
  CMakeLists.txt:3 (project)


CMake Error: CMake was unable to find a build program corresponding to "NMake Makefiles".  CMAKE_MAKE_PROGRAM is not set.  You probably need to select a different build tool.
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
-- Configuring incomplete, errors occurred!
```

Does vcpkg need a Visual Studio instance on a Windows system?

Thanks.

**Comment by Neumann-A** [comment]
Looks like your target triplet is not set and vcpkg defaults to x64-windows

**Reply by vicroms** [reply]
To be specific. Do one of the following:

* set `VCPKG_DEFAULT_TRIPLET` to `x64-mingw-static` 
* add the triplet in your command invocation `vcpkg install --triplet=x64-mingw-static`

**Comment by gitartpiano** [comment]
source: https://gist.githubusercontent.com/raspberrypisig/0e1f35019b4a930cf9df5bbac13e10ae/raw/248869a113344d72a69c46a87b8cba47b0ad0b1d/_README.txt

MSYS2 Setup
------------
1. Install MSYS2
2. Follow normal install instructions, update packages with pacman etc.
3. Add the following directories to PATH :
   C:\msys64\mingw64\bin
4. Run the following commands in a MSYS2 console:

 pacman -S --needed git base-devel mingw-w64-x86_64-toolchain  mingw-w64-x86_64-clang mingw-w64-x86_64-cmake
 cd /mingw64/bin
 ln -s mingw32-make.exe make.exe
 
 VSCode
 ---------
 1. Install VSCode
 2. Left click on Extensions on left sidebar, then search and install the C/C++ Extension Pack from Microsoft
 3. Press F1 to open the Command Palette, type:  settings json
    pick the option "Preferences: Open Workspace Settings(JSON)"
 4. [THIS IS AN OPTIONAL STEP TO HAVE MSYS2 as an integrated terminal in VSCode]
    Copy/paste the settings.json file in this gist.
    Terminal->New terminal. An MSYS2 MINGW64 terminal will open. Verify the following on the prompt:
 5. Type the following in a Windows Command Prompt to verify installation:
    g++ --version
    g++ -dumpmachine
    make --version
    cmake --version
 6. In this gist, you will find main.cpp and CMakeLists.txt. Include them in your project.
 7. Press F1 to open the Command Palette: cmake edit
    pick "CMake: Edit User-Local CMake Kits"
 8. Append to the list with a comma, the fragment from cmake-tools-kits.json in this append. Do not overwrite the entries all ready 
    there.   
 9.  Where it says "No active kit" on the CMake toolbar at the bottom of the screen. select it and you should see the entry just created
     "Mingw64 GCC 11.2.0" (if not, click Scan Kits)
 10. To build, click Build on the CMake toolbar at the bottom of the screen.
 11. To run, click the play button on the CMake toolbar at the bottom of the screen.
 
 install vcpkg in msys2 mingw64
 -------------------------------
 From 
 https://github.com/microsoft/vcpkg/blob/master/docs/users/mingw.md
 https://github.com/microsoft/vcpkg/blob/master/docs/users/integration.md
 
 1. Open MSys2 MingW64 variant of the console.
    or type:  source shell mingw64
 2.
   cd /mingw64/bin
   git clone https://github.com/microsoft/vcpkg.git
   cd vcpkg
   ./bootstrap-vcpkg.bat
   export VCPKG_DEFAULT_TRIPLET=x64-mingw-dynamic
   export VCPKG_DEFAULT_HOST_TRIPLET=x64-mingw-dynamic
   To test setup
   ./vcpkg install zlib
   /*
   This command ends with:
   The package zlib is compatible with built-in CMake targets:

    find_package(ZLIB REQUIRED)
    target_link_libraries(main PRIVATE ZLIB::ZLIB)
   */
   For the real deal
   ./vcpkg install libtcod
